go-fuzz/internal/writerset/writerset.go

```diff
@@ -0,0 +1,98 @@
+// Package writerset implements a mechanism to add and remove writers from a construct
+// similar to io.MultiWriter. Vendored from github.com/stephens2424/writerset
+package writerset
+
+import (
+	"io"
+	"net/http"
+	"sync"
+)
+
+// WriterSet wraps multiple writers like io.MultiWriter, but such that individual
+// writers are easy to add or remove as necessary.
+type WriterSet struct {
+	m  map[io.Writer]chan error
+	mu *sync.Mutex
+}
+
+// New initializes a new empty writer set.
+func New() *WriterSet {
+	return &WriterSet{
+		m:  make(map[io.Writer]chan error),
+		mu: &sync.Mutex{},
+	}
+}
+
+// Add ensures w is in the set.
+func (ws *WriterSet) Add(w io.Writer) <-chan error {
+	ws.mu.Lock()
+	defer ws.mu.Unlock()
+
+	c, ok := ws.m[w]
```

Comment:
We never add the same writer twice, so remove this part (or replace with panic).